camelConnector/src/test/java/io/vantiq/extsrc/camelconn/discover/VantiqComponentResolverTest.java

```diff
@@ -0,0 +1,736 @@
+        // expected
+        // results is presented.
+        //
+        // In this case, our MakeDigCall route uses the (dynamically loaded) dns component to make a dig call.
+        TriFunction<CamelContext, String, Object, Boolean> verifyOperation = defineVerifyOperation();
+    
+        CamelContext runnerContext = null;
+        Thread runnerThread = null;
+        CamelRunner openedRunner = null;
+        
+        try (CamelRunner runner =
+                     new CamelRunner(this.getTestMethodName(), rb, null, IVY_CACHE_PATH, DEST_PATH,
+                                     componentToInit)) {
+            openedRunner = runner;
+            runner.runRoutes(false);
+            runnerContext = runner.getCamelContext();
+            runnerThread = runner.getCamelThread();
+            assert runner.isStarted() == shouldStart;
+            if (shouldStart) {
+                // At this point, the route is running -- let's verify that it works.
+                String routeId = runnerContext.getRoutes().get(0).getId();
+                if (routeId.contains("Dig")) {
+                    if (routeId.contains("Marshaled")) {
+                        Map<String, ?> expectedMap =  Map.of("signed", MISSING_VALUE,
+                                                        "rcode", MISSING_VALUE,
+                                                        "verified", MISSING_VALUE,
+                                                        "opt", MISSING_VALUE,
+                                                        "tsig", MISSING_VALUE,
+                                                        "question", MISSING_VALUE,
+                                                        "resolver", MISSING_VALUE,
+                                                        "header", MISSING_VALUE);
+                        assert verifyOperation.apply(runnerContext, QUERY_MONKEY, expectedMap);
+                        assert verifyOperation.apply(runnerContext, QUERY_AARDVARK, expectedMap);
+                    } else {
+                        assert verifyOperation.apply(runnerContext, QUERY_MONKEY, RESPONSE_MONKEY);
+                        assert verifyOperation.apply(runnerContext, QUERY_AARDVARK, RESPONSE_AARDVARK);
+                    }
+                } else if (routeId.contains("Salesforce")) {
+                    assert verifyOperation.apply(runnerContext, "not used",
+                                                 Map.of ("totalSize", MISSING_VALUE, "done", true));
+                    assert verifyOperation.apply(runnerContext, "not used",
+                                                 Map.of ("totalSize", MISSING_VALUE, "done", true));
+                }
+            }
+        } finally {
+            while (openedRunner != null && !openedRunner.isStopped()) {
+                Thread.sleep(500);
+            }
+        }
+    
+        assertTrue("ShouldStart: " + shouldStart + ", runnerThread: " + runnerThread +
```

Comment:
If runnerThread can be null the call might fail with NPE building the first param.